community/kernel/src/main/java/org/neo4j/kernel/impl/factory/PlatformModule.java

```diff
@@ -153,8 +157,12 @@ public DependencyResolver get()
         tracers = dependencies.satisfyDependency(
                 new Tracers( desiredImplementationName, logging.getInternalLog( Tracers.class ) ) );
         dependencies.satisfyDependency( tracers.pageCacheTracer );
-        dependencies.satisfyDependency( tracers.transactionTracer );
-        dependencies.satisfyDependency( tracers.checkPointTracer );
+        TransactionTracer transactionTracer = tracers.transactionTracer;
+        dependencies.satisfyDependency(
```

Comment:
Lets please introduce a function `defaultInstance(DesiredInterface, instances...)` or smt closed to that, that will select first matching instance and will return it; that will make this kind of implementation selection much more clean